core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -121,21 +125,25 @@ class AdminClient(val time: Time,
     listAllGroupsFlattened.filter(_.protocolType == ConsumerProtocol.PROTOCOL_TYPE)
   }
 
-  def describeGroup(groupId: String): GroupSummary = {
-    val coordinator = findCoordinator(groupId)
-    val responseBody = send(coordinator, ApiKeys.DESCRIBE_GROUPS, new DescribeGroupsRequest(List(groupId).asJava))
-    val response = new DescribeGroupsResponse(responseBody)
-    val metadata = response.groups().get(groupId)
-    if (metadata == null)
-      throw new KafkaException(s"Response from broker contained no metadata for group $groupId")
-
-    Errors.forCode(metadata.errorCode()).maybeThrow()
-    val members = metadata.members().map { member =>
-      val metadata = Utils.readBytes(member.memberMetadata())
-      val assignment = Utils.readBytes(member.memberAssignment())
-      MemberSummary(member.memberId(), member.clientId(), member.clientHost(), metadata, assignment)
-    }.toList
-    GroupSummary(metadata.state(), metadata.protocolType(), metadata.protocol(), members)
+  def describeGroup(groupId: String): Option[GroupSummary] = {
```

Comment:
Oh I think it was changed because initially I had a println instead of exception in the None case in line 131 and so needed a return type for that case (which was None). I think it's safe to just have a GroupSummary as the return type of describeGroup now that the println is replaced by an exception. Updating this in next PR